utils/auth.js

```diff
@@ -1 +1,23 @@
 const jwt = require('jsonwebtoken');
+<<<<<<< HEAD
+const secretKey = process.env.SECRET_KEY;
+
+const loginReq = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization;
+
+    if (!token) {
+      return res.status(400).json({ message: 'TOKEN_NEEDED' });
+    }
+
+    const decoded = jwt.verify(token, secretKey);
+    req.userId = decoded.userId;
+    next();
```

Comment:
```suggestion
    const decoded = jwt.verify(token, process.env.SECRET_KEY);
    const user = await userService.getUserById(decoded.id)

    if (!user) {
        const error = new Error('USER_DOES_NOT_EXIST')
	error.statusCode = 404
		
	return next(error)
    }

    req.user = user;
    next();
```
인가코드에서 보다 더 안전한 보안 처리를 위해 추가적인 로직이 필요합니다.

만약 해당 사용자가 데이터베이스에 존재하지 않는다면, 즉 유효하지 않은 사용자라면 인가 에러를 발생시키고 다음 미들웨어 함수로 제어를 넘기지 않습니다. 

일종의 보안 기능을 수행하는 위 코드가 없다면, 데이터베이스에 존재하지 않는 사용자가 애플리케이션에 접근할 수 있으므로 문제가 발생할 수 있습니다.